storage/memory/memory.go

```diff
@@ -231,7 +283,12 @@ func (t *Driver) GetParentsUntilAncestor(
 	ancestor v1.DirectoryID,
 	options ...storage.Option,
 ) ([]v1.DirectoryID, error) {
-	var parents []v1.DirectoryID
+	var (
+		parentIDs []v1.DirectoryID
+		parents   []*v1.Directory
```

Comment:
We store both Parents and their IDs separately, so that we can sort the IDs based on the CreatedAt time of the parent. Only ids are returned, so instead of having to fetch the directory a second time, we can just return a slice of the parentIDs slice.